test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/BasicSearchTests.cs

```diff
@@ -476,24 +466,44 @@ public async Task GivenListOfResources_WhenSearchedWithEmptyElements_ThenAllProp
         [Trait(Traits.Priority, Priority.One)]
         public async Task GivenListOfResources_WhenSearchedWithInvalidElements_ThenOnlyValidSpecifiedPropertiesShouldBeReturned()
         {
-            const int numberOfResources = 3;
-            string[] elements = new[] { "gender", "birthDate" };
-
+            string[] elements = { "gender", "birthDate" };
             var tag = new Coding(string.Empty, Guid.NewGuid().ToString());
 
-            Patient patient = Samples.GetDefaultPatient().ToPoco<Patient>();
-            var patients = new Patient[numberOfResources];
+            Patient[] patients = await CreatePatientsWithSpecifiedElements(tag, elements);
 
-            for (int i = 0; i < numberOfResources; i++)
-            {
-                patient.Meta = new Meta();
-                patient.Meta.Tag.Add(tag);
+            Bundle bundle = await Client.SearchAsync($"Patient?_tag={tag.Code}&_elements=invalidProperty,{string.Join(',', elements)}");
 
-                FhirResponse<Patient> createdPatient = await Client.CreateAsync(patient);
-                patients[i] = MaskingNode.ForElements(new ScopedNode(createdPatient.Resource.ToTypedElement()), elements).ToPoco<Patient>();
-            }
+            ValidateBundle(bundle, patients);
+        }
 
-            Bundle bundle = await Client.SearchAsync($"Patient?_tag={tag.Code}&_elements=invalidProperty,{string.Join(',', elements)}");
+        [InlineData("id", "count")]
+        [InlineData("id,active", "true")]
+        [InlineData("id,active", "data")]
+        [InlineData("id,active", "text")]
+        [Theory]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenListOfResources_WhenSearchedWithElementsAndSummaryNotSetToFalse_ThenExceptionShouldBeThrown(string elementsValues, string summaryValue)
+        {
+            using FhirException ex = await Assert.ThrowsAsync<FhirException>(() => Client.SearchAsync($"Patient?_elements={elementsValues}&_summary={summaryValue}"));
+
+            const HttpStatusCode expectedStatusCode = HttpStatusCode.BadRequest;
+            Assert.Equal(expectedStatusCode, ex.StatusCode);
+
+            var expectedErrorMessage = $"{expectedStatusCode}: " + string.Format(Core.Resources.IncompatibleSearchParameters, KnownQueryParameterNames.Summary, KnownQueryParameterNames.Elements);
+
+            Assert.Equal(expectedErrorMessage, ex.Message);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenListOfResources_WhenSearchedWithElementsAndSummarySetToFalse_ThenOnlySpecifiedPropertiesShouldBeReturned()
```

Comment:
The `_summary` parameter could still be included in a request that has the `_elements` parameter, as long as it is set to `false`.